contracts/src/L1/rollup/ScrollChain.sol

```diff
@@ -441,56 +573,195 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
+    /// @return _totalL1MessagesPoppedInBatch The total number of L1 messages poped in this batch, including skipped one.
+    function _commitChunksV0(
+        uint256 _totalL1MessagesPoppedOverall,
+        bytes[] memory _chunks,
+        bytes calldata _skippedL1MessageBitmap
+    ) internal view returns (bytes32 _batchDataHash, uint256 _totalL1MessagesPoppedInBatch) {
+        uint256 _chunksLength = _chunks.length;
+
+        // load `batchDataHashPtr` and reserve the memory region for chunk data hashes
+        uint256 batchDataHashPtr;
+        assembly {
+            batchDataHashPtr := mload(0x40)
+            mstore(0x40, add(batchDataHashPtr, mul(_chunksLength, 32)))
+        }
+
+        // compute the data hash for each chunk
+        for (uint256 i = 0; i < _chunksLength; i++) {
+            uint256 _totalNumL1MessagesInChunk;
+            bytes32 _chunkDataHash;
+            (_chunkDataHash, _totalNumL1MessagesInChunk) = _commitChunkV0(
+                _chunks[i],
+                _totalL1MessagesPoppedInBatch,
+                _totalL1MessagesPoppedOverall,
+                _skippedL1MessageBitmap
+            );
+            unchecked {
+                _totalL1MessagesPoppedInBatch += _totalNumL1MessagesInChunk;
+                _totalL1MessagesPoppedOverall += _totalNumL1MessagesInChunk;
+            }
+            assembly {
+                mstore(batchDataHashPtr, _chunkDataHash)
+                batchDataHashPtr := add(batchDataHashPtr, 0x20)
+            }
+        }
+
+        assembly {
+            let dataLen := mul(_chunksLength, 0x20)
+            _batchDataHash := keccak256(sub(batchDataHashPtr, dataLen), dataLen)
+        }
+    }
+
+    /// @dev Internal function to commit chunks with version 1
+    /// @param _totalL1MessagesPoppedOverall The number of L1 messages popped before the list of chunks.
+    /// @param _chunks The list of chunks to commit.
+    /// @param _skippedL1MessageBitmap The bitmap indicates whether each L1 message is skipped or not.
+    /// @return _blobVersionedHash The blob versioned hash for the blob carried in this transaction.
+    /// @return _batchDataHash The computed data hash for the list of chunks.
+    /// @return _totalL1MessagesPoppedInBatch The total number of L1 messages poped in this batch, including skipped one.
+    function _commitChunksV1(
+        uint256 _totalL1MessagesPoppedOverall,
+        bytes[] memory _chunks,
+        bytes calldata _skippedL1MessageBitmap
+    )
+        internal
+        view
+        returns (
+            bytes32 _blobVersionedHash,
+            bytes32 _batchDataHash,
+            uint256 _totalL1MessagesPoppedInBatch
+        )
+    {
+        {
+            bytes32 _secondBlob;
+            // Get blob's versioned hash
+            assembly {
+                _blobVersionedHash := blobhash(0)
+                _secondBlob := blobhash(1)
+            }
+            if (_blobVersionedHash == bytes32(0)) revert ErrorNoBlobFound();
```

Comment:
Here the code only checks the `_blobVersionedHash` exists. Is it possible that the two blob hashes are different in `commitBatch` and `finalizeBatch`?

e.g., In `commitBatch` step, the sender (Sequencer here) only dumps some dummy value in the blob, while in `finalizeBatch` step, the sender (Prover here) gives the real value of blob hash.

Reply:
oh, I c, `_blobVersionedHash` is the return value, previously I did not notice it is added into `batchhash`. then no problem for me.